src/main.cpp

```diff
@@ -1242,6 +1245,93 @@ void vcRenderScene_HandlePicking(vcState *pProgramState, vcRenderData &renderDat
+              pAttributePtr += 1;
+              break;
+            case vdkAttributeTypeInfo_uint16:
+              pProgramState->udModelNodeAttributes.Set("%s = %u", pHeader->attributes.pDescriptors[i].name, *(uint16_t*)pAttributePtr);
+              pAttributePtr += 2;
+              break;
+            case vdkAttributeTypeInfo_uint32:
+              pProgramState->udModelNodeAttributes.Set("%s = %u", pHeader->attributes.pDescriptors[i].name, *(uint32_t *)pAttributePtr);
+              pAttributePtr += 4;
+              break;
+            case vdkAttributeTypeInfo_uint64:
+              pProgramState->udModelNodeAttributes.Set("%s = %" PRIu64, pHeader->attributes.pDescriptors[i].name, *(uint64_t *)pAttributePtr);
+              pAttributePtr += 8;
+              break;
+            case vdkAttributeTypeInfo_int8:
+              pProgramState->udModelNodeAttributes.Set("%s = %d", pHeader->attributes.pDescriptors[i].name, *(int8_t *)pAttributePtr);
+              pAttributePtr += 1;
+              break;
+            case vdkAttributeTypeInfo_int16:
+              pProgramState->udModelNodeAttributes.Set("%s = %d", pHeader->attributes.pDescriptors[i].name, *(int16_t *)pAttributePtr);
+              pAttributePtr += 2;
+              break;
+            case vdkAttributeTypeInfo_int32:
+              pProgramState->udModelNodeAttributes.Set("%s = %d", pHeader->attributes.pDescriptors[i].name, *(int32_t *)pAttributePtr);
+              pAttributePtr += 4;
+              break;
+            case vdkAttributeTypeInfo_int64:
+              pProgramState->udModelNodeAttributes.Set("%s = %" PRId64, pHeader->attributes.pDescriptors[i].name, *(int64_t *)pAttributePtr);
+              pAttributePtr += 8;
+              break;
+            case vdkAttributeTypeInfo_float32:
+              pProgramState->udModelNodeAttributes.Set("%s = %f", pHeader->attributes.pDescriptors[i].name, *(float *)pAttributePtr);
+              pAttributePtr += 4;
+              break;
+            case vdkAttributeTypeInfo_float64:
+              pProgramState->udModelNodeAttributes.Set("%s = %f", pHeader->attributes.pDescriptors[i].name, *(double *)pAttributePtr);
+              pAttributePtr += 8;
+              break;
+            case vdkAttributeTypeInfo_color32:
+              pProgramState->udModelNodeAttributes.Set("%s = \"RGB(%u, %u, %u)\"", pHeader->attributes.pDescriptors[i].name, pAttributePtr[3], pAttributePtr[2], pAttributePtr[1]); //BGRA internally
+              pAttributePtr += 4;
+              break;
+            case vdkAttributeTypeInfo_normal32:
+              pProgramState->udModelNodeAttributes.Set("%s = 'NORMAL'", pHeader->attributes.pDescriptors[i].name);
+              break;
+            case vdkAttributeTypeInfo_vec3f32:
+              pProgramState->udModelNodeAttributes.Set("%s = '[ %f, %f, %f ]'", pHeader->attributes.pDescriptors[i].name, *(float *)&pAttributePtr[0], *(float *)&pAttributePtr[4], *(float *)&pAttributePtr[8]); //BGRA internally
```

Comment:
You could have wrapped this and achieved more readable code, `((float*)pAttributePtr)[0]`, `[1]`, `[2]`.